rest/attachment_migration_test.go

```diff
@@ -0,0 +1,455 @@
+
+	// assert that sync info with metadata version written to the collection
+	syncInf := base.SyncInfo{}
+	_, err := ds.Get(base.SGSyncInfo, &syncInf)
+	require.NoError(t, err)
+	assert.Equal(t, "4.0", syncInf.MetaDataVersion)
+
+}
+
+// TestChangeDbCollectionsRestartMigrationJob:
+//   - Add docs before job starts, this will test that the dcp checkpoint are correctly reset upon db update later in test
+//   - Create db with collection one
+//   - Assert the attachment migration job is running
+//   - Update db config to include a new collection
+//   - Assert job runs/completes
+//   - As the job should've purged dcp collections upon new collection being added to db we expect some added docs
+//     to be processed twice in the job, so we can assert that the job has processed more docs than we added
+//   - Assert sync info: metaVersion is written to BOTH collections in the db config
+func TestChangeDbCollectionsRestartMigrationJob(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("rosmar does not support DCP client, pending CBG-4249")
+	}
+	base.TestRequiresCollections(t)
+	base.RequireNumTestDataStores(t, 2)
+	base.LongRunningTest(t)
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyAll)
+	tb := base.GetTestBucket(t)
+	rtConfig := &RestTesterConfig{
+		CustomTestBucket: tb,
+		PersistentConfig: true,
+	}
+
+	rt := NewRestTesterMultipleCollections(t, rtConfig, 2)
+	defer rt.Close()
+	ctx := base.TestCtx(t)
+	_ = rt.Bucket()
+
+	const (
+		dbName         = "db1"
+		totalDocsAdded = 8000
+	)
+
+	ds0, err := tb.GetNamedDataStore(0)
+	require.NoError(t, err)
+	ds1, err := tb.GetNamedDataStore(1)
+	require.NoError(t, err)
+	opts := &sgbucket.MutateInOptions{}
+
+	// add some docs (with xattr so they won't be ignored in the background job) to both collections
+	bodyBytes := []byte(`{"some": "body"}`)
+	for i := 0; i < 4000; i++ {
+		key := fmt.Sprintf("%s_%d", t.Name(), i)
+		xattrsInput := map[string][]byte{
+			"_xattr": []byte(`{"some":"xattr"}`),
+		}
+		_, writeErr := ds0.WriteWithXattrs(ctx, key, 0, 0, bodyBytes, xattrsInput, nil, opts)
+		require.NoError(t, writeErr)
+	}
+	for i := 0; i < 4000; i++ {
+		key := fmt.Sprintf("%s_%d", t.Name()+"greg", i)
+		xattrsInput := map[string][]byte{
+			"_xattr": []byte(`{"some":"xattr"}`),
+		}
+		_, writeErr := ds1.WriteWithXattrs(ctx, key, 0, 0, bodyBytes, xattrsInput, nil, opts)
+		require.NoError(t, writeErr)
+	}
+
+	scopesConfigC1Only := GetCollectionsConfig(t, tb, 2)
+	dataStoreNames := GetDataStoreNamesFromScopesConfig(scopesConfigC1Only)
+	scope := dataStoreNames[0].ScopeName()
+	collection1 := dataStoreNames[0].CollectionName()
+	collection2 := dataStoreNames[1].CollectionName()
+	delete(scopesConfigC1Only[scope].Collections, collection2)
+
+	scopesConfigBothCollection := GetCollectionsConfig(t, tb, 2)
+
+	// Create a db1 with one collection initially
+	dbConfig := rt.NewDbConfig()
+	dbConfig.AutoImport = false
```

Comment:
Why we setting to automatic import? Does this hurt running the tests? Do we need to run the tests with import? Just mark why the decision is made.